File: exercises/modelling_in_code/java/src/main/java/Application.java
import java.util.Objects;

public class Application {
    private ApplicationId id;
    private ApplicationReceivedDateTime receivedDateTime;

    public Application(ApplicationId id, ApplicationReceivedDateTime receivedDateTime) {
        Objects.requireNonNull(id, "ApplicationId cannot be null");
        Objects.requireNonNull(receivedDateTime, "ApplicationReceivedDateTime cannot be null");
        this.id = id;
        this.receivedDateTime = receivedDateTime;
    }

    public boolean equals(Application other) {
        return this.id.equals(other.id);
    }
}

File: exercises/modelling_in_code/typescript/src/Application.ts
import type { ApplicationId } from "./ApplicationId";
import type { ApplicationReceivedDateTime } from "./ApplicationReceivedDateTime";

export class Application {
  private id: ApplicationId;
  private receivedDateTime: ApplicationReceivedDateTime;

  constructor(id: ApplicationId, receivedDateTime: ApplicationReceivedDateTime) {
    if (!id || !receivedDateTime) {
      throw new Error("id and receivedDateTime are required");
    }
    this.id = id;
    this.receivedDateTime = receivedDateTime;
  }

  equals(other: Application) {
    return this.id.equals(other.id);
  }
}
